### PR TITLE
HTML: Add `html-vite` framework

### PR DESCRIPTION
**Pull request by GregLahaye**
Issue: N/A

## What I did
Added `html-vite` framework, along with repro templates.
I've mostly based this framework off the `web-components-vite` framework, happy to implement feedback.

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

Had trouble creating a sandbox locally as it attempts to pull `@storybook/html-vite` from the Yarn registry instead of locally when linking, but the repro works fine.

### CODE REVIEW

File: code/lib/cli/src/repro-templates.ts
Comment (IanVS) [inline code/lib/cli/src/repro-templates.ts:166]: Maybe we should include this one in CI?  @shilman WDYT?  I see that html-webpack isn't, but maybe its good to have one of the html frameworks run in CI?

**Review by IanVS — COMMENTED** [review]
This seems great to me.  I'm not sure what's up with CircleCI though...

File: code/lib/cli/src/repro-templates.ts
Comment (GregVGW) [reply code/lib/cli/src/repro-templates.ts]: I've updated `html-vite/default-ts` to run in `ci` too, seems like a good idea to me.

Comment (GregLahaye) [reply code/lib/cli/src/repro-templates.ts]: @IanVS I've updated to add `ci` to the cadence, let me know if there any other changes you'd like.

**Comment by JReinhold** [comment]
Usually there's [a manual GitHub workflow](https://github.com/storybookjs/storybook/actions/workflows/generate-repros-next.yml) maintainers can trigger pointing to a certain branch, that will generate repros from that branch, which should result in it being available in CI. However it only works for branches on the repo, not forks.
I can manually add it, but it will however get overwritten by the next scheduled repro generation from `next`, so there's a limited time window unfortunately.

cc @yannbf @kasperpeulen this PR might affect/be affected by your current CI work.

**Comment by kasperpeulen** [comment]
@JReinhold Correct, I added a comment about that.

File: code/lib/cli/src/repro-templates.ts
Comment (tmeasday) [inline code/lib/cli/src/repro-templates.ts:298]: I think we should not add the template to any CI cadences until this PR is merged, we will just test manually, then we can add it after this is merged and the template generation has run.

Comment (tmeasday) [reply code/lib/cli/src/repro-templates.ts:298]: We should probably tweak this file with these details too @yannbf : https://github.com/storybookjs/storybook/blob/next/docs/contribute/code.md#developing-a-template

**Comment by ndelangen** [comment]
I was hoping to bring this up to date with `next` branch, but got this error when trying to push:
```
> git push GregLahaye add-html-vite-framework:add-html-vite-framework
To https://github.com/GregLahaye/storybook.git
 ! [remote rejected]       add-html-vite-framework -> add-html-vite-framework (refusing to allow an OAuth App to create or update workflow `.github/workflows/trigger-circle-ci-workflow.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/GregLahaye/storybook.git'
```

File: code/frameworks/html-vite/src/index.ts
Comment (shilman) [inline code/frameworks/html-vite/src/index.ts:2]: Looks like we don't need any of these exports anymore except the `StorybookConfig` type. @IanVS any other changes we need to make to get this working?

**Review by shilman — CHANGES_REQUESTED** [review]
Looking great! One potential change

File: code/frameworks/html-vite/src/index.ts
Comment (IanVS) [reply code/frameworks/html-vite/src/index.ts:2]: Nope, you're right, we can remove the extra re-exports now.  :+1:

File: code/frameworks/html-vite/package.json
Comment (IanVS) [inline code/frameworks/html-vite/package.json:64]: We'll need to open this up to allow version 4, but we can do that in another PR, for all the vite frameworks.

**Comment by GregLahaye** [comment]
@shilman @IanVS I've removed the re-exports and bumped the version numbers.

**Comment by shilman** [comment]
Thanks @GregLahaye !!! This is a great contribution 🙌

**Comment by jonniebigodes** [comment]
@GregLahaye, thanks for this amazing contribution, one small item related to this pull request. If you could jump into our [Discord Server](https://discord.gg/storybook) and message me directly (same username)?

Looking forward to hearing from you.

Hope you have a great week.

Stay safe

**Pull request merged**
